### ISSUE TITLE
Remove panicky `HoloHash::from_raw_39` and modify downstream consumers to also not panic

### ISSUE THREAD

**Issue by 8e8b2c** [holochain/holochain#3313]
**Issue**
Calling e.g. `AgentPubKey::from_raw_39(bytes)` on something that isn't 39 bytes long causes an assertion failure. This seems surprising seeing as the function returns a `HoloHashResult`, which includes the variant `Err(HoloHashError::BadSize)`. This forces the developer to check the length proactively in addition to handling the result.

**Expected behaviour**
The function returns an `Err` instead of panicking.

**Comment by ThetaSinner** [comment]
Yes, this is unhelpful code! It's something I've run into and I'm afraid is still a TODO. Some of these hash types will accept some lengths that are known to the system like 36 and 32 bytes I believe, but then you'll get errors later.

I'll put this on my list to look at :) Thanks for raising it!

**Comment by mattyg** [comment]
In #4380 we add `try_from_*` functions for HoloHash which return errors rather than panicking.

We should just remove the panicky `from_*` functions entirely, and modify their downstream consumers to return errors as well. I'm renaming this issue to track that.
